sdk/keyvault/Azure.Security.KeyVault.Certificates/src/DeleteCertificateOperation.cs

```diff
@@ -114,5 +114,8 @@ async ValueTask<OperationState> IOperation.UpdateStateAsync(bool async, Cancella
                     return OperationState.Failure(response, new RequestFailedException(response));
             }
         }
+
+        // This method is never invoked since we don't override Operation<T>.GetRehydrationToken.
+        RehydrationToken IOperation.GetRehydrationToken() => default;
```

Comment:
@christothes As part of adding `GetRehydrationToken` to `IOperation` and `IOperation<T>`, we were forced to explicitly implement the method in a couple of libraries. @m-nash and I were on the fence about returning `default` here or throwing a `NotSupportedException`.

At the end of the day this method is not called from these libraries because the base `Operation.GetRehydrationToken` method is never overridden, but we're still wondering what would be the recommended approach here.

Reply:
throwing seems like a more obvious failure mode if, for some odd reason, this does get called at some point in the future. It also seems consistent with other operation behavior (for example accessing Value before the operation is complete).

Do you have any reasons why throwing wouldn't be a good idea?

Reply:
My only thought here is when we added GetRehydrationToken to Operation we returned null, we didn't throw.  We could have made the same choice to throw there since there were no callers at the time.

I don't have a strong opinion in either direction if both of you think we should throw lets move forward with that.

Reply:
If we do throw we should mention it in the API docs.